0011-interledger-payment-request/0011-interledger-payment-request.md

```diff
@@ -99,42 +62,17 @@ Additional binary headers, encoded as a base64 string, that should be included i
 
 It is RECOMMENDED that recipients use a Message Authentication Code (MAC) of the details of their payment request in the condition so that the condition can be used as an integrity check on incoming transfers. This allows a recipient to ensure that a transfer matches a request they previously generated, without the recipient needing to maintain a log of all outstanding payment requests.
 
-Recipients MUST include a payment request-specific component in the account ILP Address to ensure that conditions are unique per-request.
-
-### Preimage Condition from JSON
+Recipients MUST include a payment request-specific nonce in the ILP packet to ensure that conditions are unique per-request.
 
-This is how a [PREIMAGE-SHA-256 Crypto Condition](../0002-crypto-conditions) MAY be generated using an HMAC of a JSON-encoded Interledger Payment Request:
+Recipients SHOULD include an expiry date so that they do not accept incoming payments that are paid too long after their creation.
 
-1. The recipient takes the `address`, `amount`, `expires_at`, and `data` fields from the payment request. Any fields that are used for routing are probably not important to perform an integrity check upon, because they may be modified in transit and there is no further use for them once the ILP packet is received attached to an incoming transfer.
-2. The recipient uses a canonical encoding, such as [Canonical JSON](https://www.npmjs.com/package/canonical-json) to create a stringified version of `1.`
-3. The recipient uses a secret key to create an HMAC of `2.`
-4. The recipient uses the digest of `3.` as the preimage of a [PREIMAGE-SHA-256 Crypto Condition](../0002-crypto-conditions).
+### Using a PSK Implementation
 
-The following snippet shows how this can be done in JavaScript:
+It is RECOMMENDED that recipients use an implementation of the [Pre-Shared Key](../0016-pre-shared-key/0016-pre-shared-key.md) protocol to generate the packet and condition for IPR.
 
-```js
-const crypto = require('crypto') // Node.js crypto
-const stringify = require('canonical-json') // https://www.npmjs.com/package/canonical-json
-const cc = require('five-bells-condition') // https://www.npmjs.com/package/five-bells-condition
+The recipient should follow BOTH the steps normally followed by the recipient AND sender. The recipient should generate the secret in the normal fashion but MUST NOT share it. The recipient then follows the steps the sender would do in a PSK payment to create the packet and condition from the secret and the payment parameters.
```

Comment:
> The recipient should follow BOTH the steps normally followed by the recipient AND sender.

This sentence kinda makes it sound more complicated than it actually is, but I'm not sure how I would reword it.